sdk/servicebus/service-bus/src/shared/openLink.ts

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import * as log from "../log";
+import { Receiver as RheaReceiver, AwaitableSender } from "rhea-promise";
+import { translate, defaultLock } from "@azure/core-amqp";
+import { AbortSignalLike } from "@azure/core-http";
+import { AbortError } from "@azure/abort-controller";
+
+export type RheaLink = AwaitableSender | RheaReceiver;
+
+/**
+ * This is transitional code - as more code gets moved out of MessageReceiver/MessageSender
+ * we can stop passing in some of these functions.
+ *
+ * For instance
+ * - negotiateClaim + ensureTokenRenewal can be lifted out of LinkEntity
```

Comment:
out of `LinkEntity` and into where?

Reply:
Originally I was thinking just moving the methods into into openLink.ts since they're all pretty related to that activity.

However, I've been thinking about that some more. My main issue with them being here is that inheritance can make it difficult to see the proper separation between LinkEntity and its child classes. So when I was tracing through this it wasn't entirely obvious which parts were "internal" state to LinkEntity, etc...

Having been through it more I could see it staying but it'd be nice to make that first-time read a bit easier.